shared/src/http.rs

```diff
@@ -1,20 +0,0 @@
-use std::time::Duration;
-
-use anyhow::Result;
-use reqwest::{
-    header::{HeaderValue, ACCEPT},
-    Client, ClientBuilder,
-};
-
-pub fn default_http_client() -> Result<Client> {
-    Ok(ClientBuilder::new()
-        .user_agent("gp-v2-services/2.0.0")
-        .default_headers(
-            vec![(ACCEPT, HeaderValue::from_static("application/json"))]
-                .into_iter()
-                .collect(),
-        )
-        .timeout(Duration::from_secs(60))
-        .https_only(true)
```

Comment:
I have reused the same user agent for the client we create in main. I have not reused the header and https_only because I do not believe that they are needed. We didn't discuss those in the PR that introduced this function and all of the solvers that used to use this function (paraswap, matcha, 1nch) still pass their #ignored tests that talk to the real api.